SECP-service/src/test/java/com/visucius/secp/resources/EntryResourceTest.java

```diff
@@ -0,0 +1,100 @@
+    private TokenController tokenController = Mockito.mock((TokenController.class));
+    private UserRegistrationController userRegistrationController  = Mockito.mock((UserRegistrationController.class));
+    private LoginRequestController loginRequestController = new LoginRequestController(tokenController, userDAO);
+
+
+    @Rule
+    public final ResourceTestRule resources = ResourceTestRule.builder()
+        .addResource(new EntryResource(userRegistrationController, loginRequestController))
+        .build();
+
+    @Test
+    public void testLoginWithNoArgument() {
+        Response response = resources.client().target(url).request().post(null);
+        ResponseValidator.validate(response, 400, "Login Failed. Please provide your username and password.");
+    }
+
+    @Test
+    public void testLoginWithNoUsername() {
+        LoginRequestDTO loginRequestDTO = new LoginRequestDTO();
+        loginRequestDTO.setPassword("password");
+        Response response = resources.client().target(url).request().post(Entity.json(loginRequestDTO));
+        ResponseValidator.validate(response, 400, "Login Failed. Please provide your username.");
+    }
+
+    @Test
+    public void testLoginWithNoPassword() {
+        LoginRequestDTO loginRequestDTO = new LoginRequestDTO();
+        loginRequestDTO.setUsername("username");
+        Response response = resources.client().target(url).request().post(Entity.json(loginRequestDTO));
+        ResponseValidator.validate(response, 400, "Login Failed. Please provide your password.");
+    }
+
+    @Test
+    public void testLoginWithInvalidUsername() {
+        LoginRequestDTO loginRequestDTO = new LoginRequestDTO();
+        loginRequestDTO.setUsername("username");
+        loginRequestDTO.setPassword("password");
+
+        Mockito.when(userDAO.findByUserName("duplicateUsername")).thenReturn(new User());
+
+        Response response = resources.client().target(url).request().post(Entity.json(loginRequestDTO));
+        ResponseValidator.validate(response, 401, "Login Failed. username does not exist.");
+    }
+
+    @Test
+    public void testLoginWithInvalidPassword() throws Exception {
+        LoginRequestDTO loginRequestDTO = new LoginRequestDTO();
+        loginRequestDTO.setUsername("username");
+        loginRequestDTO.setPassword("password");
+
+        User mockedUser = new User();
+        mockedUser.setUsername("username");
+        mockedUser.setPassword(PasswordUtil.createHash("12345"));
+        Mockito.when(userDAO.findByUserName("username")).thenReturn(mockedUser);
+
+        Response response = resources.client().target(url).request().post(Entity.json(loginRequestDTO));
+        ResponseValidator.validate(response, 401, "Login Failed. Incorrect password.");
+    }
+
+    @Test
+    public void testLoginSuccess() throws Exception {
+        LoginRequestDTO loginRequestDTO = new LoginRequestDTO();
+        loginRequestDTO.setUsername("username");
```

Comment:
create a method to set username and password so your not copying code

Reply:
I agree.